chapter03/README.md

```diff
@@ -55,15 +53,23 @@ the stack of symbol tables, in lexical order.
 
 ## ScopeNode
 
-We wrap the stack of symbol tables in a `ScopeNode`, and the symbol table's
-`Integer` is just an index in the `ScopeNode`.  When a name binding is added to
-a symbol table, we also make the newly added node an input in the `ScopeNode`.
-This means that the `ScopeNode` is a user of all nodes that define new names -
-crucial, because the normal peepholes will remove dead Nodes and every new Node
-is effectively dead at the instance of creation.
+A ScopeNode encapsulates the stack of symbol tables that are used to implement lexical scopes.
 
-When a scope is exited, we not only pop the symbol table representing the
-scope, but also remove the relevant input nodes.  If the variable's expression
+* The nodes referenced by the names in a lexical scope become inputs to the ScopeNode; this makes ScopeNode a user of these values, thus keeping them alive while a particular lexical scope is alive. 
+  When the lexical scope ends, those inputs are removed. This is crucial because peepholes will remove dead Nodes and every new Node is effectively dead at the instance of creation without such a reference.
+* In [Chapter 5](../chapter05/README.md) when we implement branches, the ScopeNode gets duplicated at a branch point and then get merged when the branch joins. This allows 
+  the names to evolve in each branch independently, and at the merge point, names that have diverged, require a Phi node. 
```

Comment:
Split runon sentence, drop extra commas
"evolve in each branch independently.  At the merge point names that have diverged require a Phi node."